### PR TITLE
Send work item failed progress messages

### LINKED ISSUE #76: Send work item failed progress messages
We don't send a progress message when a work item fails. What should the status be?

### PR DESCRIPTION
**Pull request by akshay-zz**
Fix #76

### CODE REVIEW

File: tests/Foundatio.Tests/Jobs/WorkItemJobTests.cs [hunk 1]
```diff
@@ -246,7 +246,7 @@ public class WorkItemJobTests : TestWithLoggingBase {
 
             await job.RunUntilEmptyAsync();
             await countdown.WaitAsync(TimeSpan.FromSeconds(2));
-            Assert.Equal(0, countdown.CurrentCount);
```
Comment (niemyjski) [inline tests/Foundatio.Tests/Jobs/WorkItemJobTests.cs:249]: We should keep this assert at 0 and update the `var countdown = new AsyncCountdownEvent(2);`  to a value of 2.

File: src/Foundatio/Jobs/WorkItemJob/WorkItemJob.cs [hunk 2]
```diff
@@ -132,6 +132,9 @@ public class WorkItemJob : IQueueJob<WorkItemData>, IHaveLogger {
 
                 return JobResult.Success;
             } catch (Exception ex) {
+
+                await ReportProgressAsync(handler, queueEntry, -1, $"Failed: {ex.Message}").AnyContext();
```
Comment (niemyjski) [inline src/Foundatio/Jobs/WorkItemJob/WorkItemJob.cs:136]:
```suggestion
            if (queueEntry.Value.SendProgressReports)
                await ReportProgressAsync(handler, queueEntry, -1, ex.Message).AnyContext();
```

We should probably respect the `SendProgressReports` property on the queue item. I just noticed there is another spot where we aren't looking at this `SendProgressReports` flag in the `progressCallback `.

@ejsmith any concerns around try catching this here? I don't really like nesting this but if this fails, we won't abandon or release locks.

Comment (ejsmith) [reply src/Foundatio/Jobs/WorkItemJob/WorkItemJob.cs:136]: We should respect the setting for sure. Try catch can't hurt.

**Review by ejsmith — CHANGES_REQUESTED** [review]
Added some feedback

File: tests/Foundatio.Tests/Jobs/WorkItemJobTests.cs (hunk 1 above)
Comment (akshay-zz) [reply tests/Foundatio.Tests/Jobs/WorkItemJobTests.cs]: done

File: src/Foundatio/Jobs/WorkItemJob/WorkItemJob.cs (hunk 2 above)
Comment (akshay-zz) [reply src/Foundatio/Jobs/WorkItemJob/WorkItemJob.cs]: Added flag `SendProgressReports`

**Review by ejsmith — APPROVED** [review]
LGTM

**Review by niemyjski — APPROVED** [review]

**Comment by niemyjski** [comment]
Thanks for the PR!

**Pull request merged**